sycl/include/sycl/handler.hpp

```diff
@@ -1829,12 +1829,12 @@ class __SYCL_EXPORT handler {
   void parallel_for(range<Dims> NumWorkItems, id<Dims> WorkItemOffset,
                     _KERNELFUNCPARAM(KernelFunc)) {
     throwIfActionIsCreated();
-#if defined(__SYCL_SPIR_DEVICE__) || !defined(__SYCL_DEVICE_ONLY__)
+// #if defined(__SYCL_SPIR_DEVICE__) || !defined(__SYCL_DEVICE_ONLY__)
 #if !defined(__SYCL_DEVICE_ONLY__)
     if (detail::getDeviceFromHandler(*this).has(
             aspect::ext_xilinx_single_task_only)) {
 #endif
-      single_task<detail::name_wrapper<KernelName>>(
+      single_task<KernelName>(
```

Comment:
While you are here it would be nice to add a comment explaining why we are having this single task and `detail::serialize_parallel_for` here.